### ISSUE TITLE
Disable autoupdate feature

### ISSUE THREAD

**Issue by pphaneuf** [quadra-game/quadra#86]

**Comment by pphaneuf** [comment]
I was surprised to see in the quadra source code that there is an
auto-update feature (didn't see anything about it in the docs) that
connects to code.google.com to check for updates.

There are some people who don't like when software &quot;phone home&quot; without
their consent, so it would be nice if there was a command line switch or an
option in the menus to disable it.

Also, it looks like the auto-update is executed upon startup. It happens
that my connection sometimes gets slow (bad wlan reception) so quadra
hangs/lags on startup.

**Comment by pphaneuf** [comment]
You found documentation? Which one? :-)

There's actually a compile-time option to disable this feature (which is more an
update notification than an actual auto-update, since it does nothing to actually
update). I suppose it could be a little bit easier, but I would actually expect most
distributions to turn it off, since they will have their own package updating system.

It also sends just about no information at all to the remote server (there's the
version string in the User-Agent, that's about it), and we don't even have access to
the web server logs (it's getting it straight from the WebDAV/SVN repository).
There's a bit more information sent when playing on the Internet actually (the SDL
video driver, if I remember, things like that, because we wanted to know how much
people care about Linux vs Mac vs Windows).

The update check is at startup indeed, but it is asynchronous. I tested it with a
sabotaged Internet connection, and it gets to the main menu just as quickly as
without it, but you won't see the "update available" message until the next time you
restart the game if the answer wasn't available before we got to the menu. If that is
broken, we should definitely fix it.

This feature (and the way to disable it) should definitely be documented better.
